libyang/data.py

```diff
@@ -184,7 +184,7 @@ def find_all(self, xpath: str) -> Iterator["DNode"]:
             raise self.context.error("cannot find path")
         try:
             for i in range(node_set.number):
-                yield DNode.new(self.context, node_set.d[i])
+                yield DNode.new(self.context, node_set.set.s[i])
```

Comment:
We are dealing with data nodes here, `node_set.set.s[i]` is a `struct lys_node *`. I guess it works because the pointer is cast systematically in `DNode.new` but it would be better to use `node_set.set.d[i]` instead.

However, this structure field is not exported in `cdefs.h`. You'll need to add it.

Reply:
Good point. I added `d` to the struct.